core/src/main/kotlin/behavior/MessageBehavior.kt

```diff
@@ -178,20 +199,22 @@ interface MessageBehavior : KordEntity, Strategizable {
      * Returns a new [MessageBehavior] with the given [strategy].
      */
     override fun withStrategy(
-            strategy: EntitySupplyStrategy<*>,
-    ): MessageBehavior = MessageBehavior(channelId, id, kord, strategy)
+        strategy: EntitySupplyStrategy<*>,
+    ): MessageBehavior = MessageBehavior(channelId, id, kord, webhookId, strategy)
 
 }
 
- fun MessageBehavior(
+fun MessageBehavior(
     channelId: Snowflake,
     messageId: Snowflake,
     kord: Kord,
+    webhookId: Snowflake? = null,
```

Comment:
What else should I do then?
Editing webhook messages require a different endpoint, which also requires the webhook id, hence it's part of the endpoint path. If the message is a webhook message but does not have a webhook id because discord didn't give it to us the edit would fail either way. Ideally we would have a webhookmessage entity but that raises the issue again that we don't always know whether something is a webhook message or not

Reply:
> What else should I do then?

Make it a mandatory argument when editing a message from a behavior, drop the field from the behavior interface.

Discord has different behavior based on message types, so it might be time to implement Discord's message types as concrete types ourselves. If we do, we can create an overload for webhook messages that doesn't require the id argument. That's something for another PR though.